python_modules/dagster/dagster/_config/structured_config/resource_verification.py

```diff
@@ -0,0 +1,33 @@
+from enum import Enum
+from typing import NamedTuple, Optional
+
+from dagster._serdes.serdes import whitelist_for_serdes
+
+
+@whitelist_for_serdes
+class VerificationStatus(Enum):
+    SUCCESS = "SUCCESS"
+    FAILURE = "FAILURE"
+
+
+@whitelist_for_serdes
+class VerificationResult(NamedTuple):
+    status: VerificationStatus
+    message: Optional[str]
+
+    @classmethod
+    def success(cls, message: Optional[str] = None):
+        """Create a successful verification result.
+        """
+        return cls(VerificationStatus.SUCCESS, message)
+
+    @classmethod
+    def failure(cls, message: Optional[str] = None):
+        """Create a failed verification result.
+        """
+        return cls(VerificationStatus.FAILURE, message)
```

Comment:
bike shed territory but did you consider OK result & `Exception`  instead of OK and Failure (and handling Exception) ?

Reply:
I'm not married to this approach, but we do handle the exception case in the callsites (it gets turned into a failure in the UI etc). The nice thing about this approach w/ runs is there's a distinction between "unable to connect" and "error in connection code," which are two distinct failstates.